crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1443,6 +1443,14 @@ impl<'db> Type<'db> {
         }
     }
 
+    pub(crate) fn is_exception_class(self, db: &'db dyn Db) -> bool {
```

Comment:
This method also returns `true` (and is correct to do so) for `Type::Any`, `Type::Unknown` and `Type::Todo`, so if we're going to keep it for now, it would be better to give it a more precise name than `is_exception_class`:

```suggestion
    pub(crate) fn is_assignable_to_subclass_of_base_exception(self, db: &'db dyn Db) -> bool {
```

But I would still favour just simplifying the signature of `KnownClass::to_subclass_of()` so that we don't need this method at all